crates/adapters/src/format/avro/test.rs

```diff
@@ -319,28 +351,222 @@ fn test_raw_avro_parser() {
+        .collect::<Vec<_>>();
+    let expected_output = vals
+        .iter()
+        .map(|v| {
+            let mut v = v.clone();
+            // set missing field to NULL
+            v.field_0 = None;
+            MockUpdate::Insert(v)
+        })
+        .collect::<Vec<_>>();
+
+    let test = TestCase {
+        relation_schema: TestStruct2::relation_schema(),
+        config: AvroParserConfig {
+            update_format: AvroUpdateFormat::Raw,
+            schema: Some(schema_str.to_string()),
+            no_schema_id: false,
+            registry_config: Default::default(),
+        },
+        input_batches,
+        expected_output,
+    };
+
+    run_parser_test(vec![test]);
+}
+
+/// Deserializing non-optional fields into NULL-able columns.
+#[test]
+fn test_non_null_to_nullable() {
+    // Make `name` column non-optional.
+    let schema_str = r#"{
+        "type": "record",
+        "name": "TestStruct2",
+        "connect.name": "test_namespace.TestStruct2",
+        "fields": [
+            { "name": "id", "type": "long" },
+            { "name": "name", "type": "string" },
+            { "name": "b", "type": "boolean" },
+            { "name": "ts", "type": "long", "logicalType": "timestamp-micros" },
+            { "name": "dt", "type": "int", "logicalType": "date" },
+            {
+                "name": "es",
+                "type":
+                    {
+                        "type": "record",
+                        "name": "EmbeddedStruct",
+                        "fields": [
+                            { "name": "a", "type": "boolean" }
+                        ]
+                    }
+            },
+            {
+                "name": "m",
+                "type":
+                    {
+                        "type": "map",
+                        "values": "long"
+                    }
+            }
+        ]
+    }"#;
+
+    let schema = AvroSchema::parse_str(&schema_str).unwrap();
+    let vals = vec![TestStruct2 {
+        field: 1,
+        field_0: Some("test".to_string()),
+        ..Default::default()
+    }];
+    let input_batches = vals
+        .iter()
+        .map(|v| (serialize_record(v, &schema), vec![]))
+        .collect::<Vec<_>>();
+    let expected_output = vals
+        .iter()
+        .map(|v| MockUpdate::Insert(v.clone()))
+        .collect::<Vec<_>>();
+
+    let test = TestCase {
+        relation_schema: TestStruct2::relation_schema(),
+        config: AvroParserConfig {
+            update_format: AvroUpdateFormat::Raw,
+            schema: Some(schema_str.to_string()),
+            no_schema_id: false,
+            registry_config: Default::default(),
+        },
+        input_batches,
+        expected_output,
+    };
+
+    run_parser_test(vec![test]);
+}
+
+/// Deserializing non-optional fields into NULL-able columns.
+#[test]
+fn test_ms_time_nullable() {
+    // timestamp-millis instead of timestamp-micros
```

Comment:
should this comment replace the one above?

Reply:
oops